sdk-base/src/main/java/com/mapbox/maps/threading/AnimationThreadController.kt

```diff
@@ -0,0 +1,115 @@
+package com.mapbox.maps.threading
+
+import android.os.Handler
+import android.os.HandlerThread
+import android.os.Looper
+import com.mapbox.maps.MapboxExperimental
+import com.mapbox.maps.logW
+
+/**
+ * Experimental controller singleton allowing to use background thread for all Mapbox animations.
+ * Most likely should not be used directly, calling [useBackgroundThread] should do all the work.
+ *
+ * By default all Mapbox animations are running on Android Main thread.
+ *
+ * Following Mapbox plugins are updated to make use of background animation thread:
+ *  - plugin-locationcomponent
+ *  - plugin-animation
+ *  - plugin-viewport
+ *
+ *  Switching between [useBackgroundThread] and [useMainThread] could be done at any point in runtime.
+ *
+ *  Using this class is extremely unsafe and error prone, it should be used in very specific use-cases!
+ *  Even when it's essential to use it (for example Android Auto use-case on specific devices when locking the device)
+ *  it would be safer to use Android Main thread as much as possible and switch to background thread
+ *  only when it's needed.
+ */
+@MapboxExperimental
+object AnimationThreadController {
+
+  /**
+   * Indicates if background thread is used now to run Mapbox animators.
+   */
+  var usingBackgroundThread = false
+    private set
+  private val mainHandler = Handler(Looper.getMainLooper())
+
+  private var backgroundAnimationThread: HandlerThread? = null
+  private var backgroundAnimationHandler: Handler? = null
+
+  /**
+   * Spawn new animation [HandlerThread] that will be used for all Mapbox animators.
+   * If animation thread is already running - function is no-op.
+   *
+   * Behaviour is undefined if any animators are running when this method is called.
+   */
+  @Synchronized
+  fun useBackgroundThread() {
+    if (usingBackgroundThread) {
+      return
+    }
+    usingBackgroundThread = true
+    backgroundAnimationThread?.quit()
+    backgroundAnimationThread = HandlerThread(BACKGROUND_ANIMATION_THREAD_NAME).apply {
+      start()
+      backgroundAnimationHandler = Handler(looper)
+    }
+  }
+
+  /**
+   * Execute [function] on Android Main thread.
+   *
+   * If called from Android Main thread - will be executed immediately in the same callchain
+   * otherwise will be posted on Android Main thread explicitly.
+   */
+  @Synchronized
+  fun postOnMainThread(function: () -> Unit) {
+    if (Looper.myLooper() == Looper.getMainLooper()) {
+      function.invoke()
```

Comment:
Since this method is marked as synchronized then it means that any other method calling `postOnMainThread` will block for as long as this `function.invoke` takes.

Is that ok? It sounds a bit risky to me to let external functions block this.

Reply:
thanks for the interesting offline discussion!
decided to remove synchronized and add volatile modifier for the bool flag